durablefunctionsmonitor.dotnetisolated.core/Common/Auth.cs

```diff
@@ -334,6 +324,68 @@ private static string TryGetHubNameFromHostJson()
             }
         }
 
+        private static async Task<ClaimsPrincipal> GetClaimsPrincipal(HttpRequestData request, DfmSettings settings)
+        {
+            // First trying the request object
+            var easyAuthPrincipal = new ClaimsPrincipal(request.Identities);
+
+            if (easyAuthPrincipal.Identity != null && easyAuthPrincipal.Identity.IsAuthenticated && easyAuthPrincipal.HasClaim(c => c.Type == settings.UserNameClaimName))
+            {
+                // Authentication was done by EasyAuth and converted into a ClaimsPrincipal by the Functions runtime. Returning immediately.
+                return easyAuthPrincipal;
+            }
+
+            // Then trying to parse the x-ms-client-principal header
+            if (request.Headers.TryGetValues("x-ms-client-principal", out var clientPrincipalHeaderValues))
+            {
+                return ParseMsClientPrincipalHeader(clientPrincipalHeaderValues.Single(), settings);
+            }
+
+            // Validating and parsing the access token ourselves
+            request.Headers.TryGetValues("Authorization", out var authHeaderValues);
+            return await ValidateToken(authHeaderValues?.SingleOrDefault());
+        }
+
+        private  static ClaimsPrincipal ParseMsClientPrincipalHeader(string headerValue, DfmSettings settings)
```

Comment:
Probably worth adding a link to the docs
```suggestion
        // parsing header directly as per the guidance here: https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-http-webhook-trigger?tabs=python-v2%2Cisolated-process%2Cnodejs-v4%2Cfunctionsv2&pivots=programming-language-csharp#working-with-client-identities
        private  static ClaimsPrincipal ParseMsClientPrincipalHeader(string headerValue, DfmSettings settings)
```